lib/logger.js

```diff
@@ -0,0 +1,56 @@
+'use strict'
+
+const {createLogger} = require('@logdna/logger')
+
+const {BUCKET_NAME_REGEX, HOSTNAME_REGEX} = require('./constants.js')
+const config = require('./config.js')
+
+let logger
+
+module.exports = {
+  buildLoggerURL
+, createLoggerClient
+}
+
+function buildLoggerURL(config) {
+  const ssl = config.get('ssl')
+  const host = config.get('ingestion-host')
+  const port = config.get('ingestion-port')
+  const endpoint = config.get('ingestion-endpoint')
+  const protocol = ssl ? 'https' : 'http'
+  const url = `${protocol}://${host}:${port}${endpoint}`
+  return url
+}
+
+function createLoggerClient(bucket) {
+  config.validateEnvVars()
+  let tags = config.get('tags')
+  if (tags) {
+    tags = tags.split(',').map((tag) => {
+      return tag.trim()
+    }).filter(Boolean).join(',')
+  }
+
+  config.set('tags', tags)
+  let hostname = config.get('hostname')
+  if (!hostname && bucket) {
+    hostname = bucket.replace(BUCKET_NAME_REGEX, '_')
+  }
+
+  if (hostname) {
+    config.set('hostname', hostname.replace(HOSTNAME_REGEX, ''))
+  }
```

Comment:
any reason this needs to be here? feels like it should be done outside of this function, in `index.js`. instantiating the logger shouldn't affect `config`, it should just use it.